api/api.go

```diff
@@ -93,6 +106,24 @@ func (api *API) WriteJSON(w http.ResponseWriter, obj interface{}) {
 	}
 }
 
+// WriteJSONCustomStatus writes the object to the ResponseWriter. If the
+// encoding fails, an error is written instead. The Content-Type of the response
+// header is set accordingly.
+func (api *API) WriteJSONCustomStatus(w http.ResponseWriter, obj interface{}, status int) {
+	w.Header().Set("Content-Type", "application/json; charset=utf-8")
+	w.WriteHeader(status)
+	api.staticLogger.Traceln(status)
+	err := json.NewEncoder(w).Encode(obj)
+	if err != nil {
+		api.staticLogger.Debugln(err)
+	}
+	if _, isJSONErr := err.(*json.SyntaxError); isJSONErr {
```

Comment:
does this not panic if err is `nil`?

Reply:
No, the type assertion only panics if you don't read its "ok" value, in this case `isJSONErr`. I believe using an assertion on a `nil` value works just like a method call on `nil` receiver - as long as the method doesn't access the receiver there is no issue. Playing around with this empirically proved those assumptions.